spacewalk/certs-tools/rhn_bootstrap_strings.py

```diff
@@ -555,6 +553,93 @@ def getRegistrationStackSh(saltEnabled):
   zypper --non-interactive up {PKG_NAME_UPDATE} $RHNLIB_PKG ||:
 fi
 
+if [ "$INSTALLER" == apt ]; then
+    function check_deb_pkg_installed {{
+        dpkg-query -W -f='${{Status}}' $1 2>/dev/null | grep -q "ok installed"
+    }}
+
+    function getA_CLIENT_CODE_BASE() {{
+        local BASE=""
+        local VERSION=""
+
+	if [ -f /etc/os-release ]; then
+	    BASE=$(source /etc/os-release; echo $ID)
+	    VERSION=$(source /etc/os-release; echo $VERSION_ID)
+        fi
+        A_CLIENT_CODE_BASE="${{BASE:-unknown}}"
+	local VERCOMPS=(${{VERSION/\./ }}) # split into an array 18.04 -> (18 04)
+        A_CLIENT_CODE_MAJOR_VERSION=${{VERCOMPS[0]}}
+	A_CLIENT_CODE_MINOR_VERSION=$((${{VERCOMPS[1]}} + 0)) # convert "04" -> 4
+    }}
+
+    function getA_MISSING() {{
+        local NEEDED="salt-common salt-minion"
+        A_MISSING=""
+        for P in $NEEDED; do
+            check_deb_pkg_installed "$P" || A_MISSING="$A_MISSING $P"
+        done
+    }}
+
+    function test_deb_repo_exists() {{
+      local repourl="$CLIENT_REPO_URL"
+
+      $FETCH $repourl/dists/bootstrap/Release
```

Comment:
Hmm, does this tool generate such a deep structure for the Release file?

Do we use the same path also for the bootstrap salt state ?

Reply:
Yes, the structure of a Debian repo is quite deep. The `/dists/bootstrap/Release` part is typically handled by `apt` under the hood but in this case we want to check if the repo exists, hence the deep url.